tests/net/socket/tcp/src/main.c

```diff
@@ -723,6 +723,45 @@ void test_open_close_immediately(void)
 	k_sleep(TCP_TEARDOWN_TIMEOUT);
 }
 
+void test_open_obstructed(void)
```

Comment:
I think we could use more descriptive name, simply `test_connect_timeout()` is more specific IMO.